spanner/spanner_samples.rb

```diff
@@ -0,0 +1,391 @@
+      { SingerId: 3, FirstName: "Alice",    LastName: "Trentor" },
+      { SingerId: 4, FirstName: "Lea",      LastName: "Martin" },
+      { SingerId: 5, FirstName: "David",    LastName: "Lomond" }
+    ]
+    c.insert "Albums", [
+      { SingerId: 1, AlbumId: 1, AlbumTitle: "Go, Go, Go" },
+      { SingerId: 1, AlbumId: 2, AlbumTitle: "Total Junk" },
+      { SingerId: 2, AlbumId: 1, AlbumTitle: "Green" },
+      { SingerId: 2, AlbumId: 2, AlbumTitle: "Forever Hold your Peace" },
+      { SingerId: 2, AlbumId: 3, AlbumTitle: "Terrified" }
+    ]
+  end
+
+  puts "Inserted data"
+  # [END insert_data]
+end
+
+def query_data project_id:, instance_id:, database_id:
+  # [START query_data]
+  # project_id  = "Your Google Cloud project ID"
+  # instance_id = "Your Spanner instance ID"
+  # database_id = "Your Spanner database ID"
+
+  require "google/cloud/spanner"
+
+  spanner = Google::Cloud::Spanner.new project: project_id
+  client  = spanner.client instance_id, database_id
+
+  client.execute("SELECT SingerId, AlbumId, AlbumTitle FROM Albums").rows.each do |row|
+    puts "#{row[:SingerId]} #{row[:AlbumId]} #{row[:AlbumTitle]}"
+  end
+  # [END query_data]
+end
+
+def read_data project_id:, instance_id:, database_id:
+  # [START read_data]
+  # project_id  = "Your Google Cloud project ID"
+  # instance_id = "Your Spanner instance ID"
+  # database_id = "Your Spanner database ID"
+
+  require "google/cloud/spanner"
+
+  spanner = Google::Cloud::Spanner.new project: project_id
+  client  = spanner.client instance_id, database_id
+
+  client.read("Albums", [:SingerId, :AlbumId, :AlbumTitle]).rows.each do |row|
+    puts "#{row[:SingerId]} #{row[:AlbumId]} #{row[:AlbumTitle]}"
+  end
+  # [END read_data]
+end
+
+def create_index project_id:, instance_id:, database_id:
+  # [START create_index]
+  # project_id  = "Your Google Cloud project ID"
+  # instance_id = "Your Spanner instance ID"
+  # database_id = "Your Spanner database ID"
+
+  require "google/cloud/spanner"
+
+  spanner  = Google::Cloud::Spanner.new project: project_id
+  instance = spanner.instance instance_id
+  database = instance.database database_id
+
+  job = database.update statements: [
+    "CREATE INDEX AlbumsByAlbumTitle ON Albums(AlbumTitle)"
+  ]
+
+  puts "Waiting for database update to complete"
+
+  job.wait_until_done!
+
+  puts "Added the AlbumsByAlbumTitle index"
+  # [END create_index]
+end
+
+def create_storing_index project_id:, instance_id:, database_id:
+  # [START create_storing_index]
+  # project_id  = "Your Google Cloud project ID"
+  # instance_id = "Your Spanner instance ID"
+  # database_id = "Your Spanner database ID"
+
+  require "google/cloud/spanner"
+
+  spanner  = Google::Cloud::Spanner.new project: project_id
+  instance = spanner.instance instance_id
+  database = instance.database database_id
+
+  job = database.update statements: [
+    "CREATE INDEX AlbumsByAlbumTitle2 ON Albums(AlbumTitle) " +
```

Comment:
Why not remove the `" +`? Referencing line 27, it's a multi-line constant string.